controllers/ptpinterface_controller.go

```diff
@@ -674,3 +691,84 @@ func (r *PtpInterfaceReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if ok {
+		restoreStatus := &cloudManager.RestoreStatus{}
+		err := json.Unmarshal([]byte(config), &restoreStatus)
+		if err == nil {
+			if restoreStatus.InSync != nil {
+				instance.Status.InSync = *restoreStatus.InSync
+			}
+			if restoreStatus.DeploymentScope != nil {
+				if strings.ToLower(*restoreStatus.DeploymentScope) == "bootstrap" ||
+					strings.ToLower(*restoreStatus.DeploymentScope) == "principal" {
+					instance.Status.DeploymentScope = *restoreStatus.DeploymentScope
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid deploymentScope value '%s', scope will be set to boostrap", *restoreStatus.DeploymentScope)
+					instance.Status.DeploymentScope = "bootstrap"
+				}
+			} else {
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "The 'deploymentScope' is not specified, assuming 'bootstrap' scope.")
+				instance.Status.DeploymentScope = "bootstrap"
+			}
+			if restoreStatus.StrategyRequired != nil {
+				if *restoreStatus.StrategyRequired == "not_required" ||
+					*restoreStatus.StrategyRequired == "lock_required" ||
+					*restoreStatus.StrategyRequired == "unlock_required" {
+					instance.Status.StrategyRequired = *restoreStatus.StrategyRequired
+				} else {
+					r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Invalid strategyRequired value '%s', it will be set to 'not_required'", *restoreStatus.StrategyRequired)
+					instance.Status.StrategyRequired = "not_required"
+				}
+			} else {
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "The 'strategyRequired' is not specified, assuming 'not_required'.")
+				instance.Status.StrategyRequired = "not_required"
+			}
+			instance.Status.Reconciled = true
+			instance.Status.ObservedGeneration = instance.ObjectMeta.Generation
+			err = r.Client.Status().Update(context.TODO(), instance)
+			if err != nil {
+				log_err_msg := fmt.Sprintf(
+					"Failed to update ptp interface status while restoring '%s' resource. Error: %s",
+					instance.Name,
+					err)
+				return common.NewResourceStatusDependency(log_err_msg)
+			} else {
+				StatusUpdate := fmt.Sprintf("Status updated for PtpInterface resource '%s' during restore with following values: Reconciled=%t InSync=%t DeploymentScope=%s",
+					instance.Name, instance.Status.Reconciled, instance.Status.InSync, instance.Status.DeploymentScope)
+				r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, StatusUpdate)
+
+			}
+		} else {
+			r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Failed to unmarshal '%s'", err)
+		}
+	}
+	return nil
+}
+
+// Clear annotation RestoreInProgress
+func (r *PtpInterfaceReconciler) ClearAnnotation(instance *starlingxv1.PtpInterface) error {
+	delete(instance.Annotations, cloudManager.RestoreInProgress)
+	if !utils.ContainsString(instance.ObjectMeta.Finalizers, PtpInterfaceFinalizerName) {
```

Comment:
Why to add a finalizer here?

Reply:
In case of restore-in-progress annotation, it would set the deploymentScope, inSync, strategyRequired and reconciled status and end the reconciliation (no finalizer is added). We are adding it as part of restore-in-progress just in case user wants to delete an object and the DM has to handle it (in case of supported objects).